test/general/signature.js

```diff
@@ -478,41 +478,52 @@ vwjE8mqJXetNMfj8r2SCyvkEnlVRYR+/mnge+ib56FdJ8uKtqSxyvgA=
 -----END PGP MESSAGE-----`;
 
   it('Testing signature checking on CAST5-enciphered message', async function() {
-    const priv_key = (await openpgp.key.readArmored(priv_key_arm1)).keys[0];
-    const pub_key = (await openpgp.key.readArmored(pub_key_arm1)).keys[0];
-    const msg = await openpgp.message.readArmored(msg_arm1);
-    await priv_key.decrypt("abcd");
-    return openpgp.decrypt({ privateKeys: priv_key, publicKeys:[pub_key], message:msg }).then(function(decrypted) {
+    const { reject_message_hash_algorithms } = openpgp.config;
+    Object.assign(openpgp.config, { reject_message_hash_algorithms: new Set([openpgp.enums.hash.md5, openpgp.enums.hash.ripemd]) });
+    try {
```

Comment:
Why use try in tests? It makes tests more confusing in a way.

Reply:
The `try/finally` is to make sure that the config gets restored if the test fails (throws).

Maybe since it's happening a lot we could organize it in a bit more structured way (e.g. with `beforeEach`/`afterEach`, as is already done in some places) but when it's just for some of the tests I think it's still fairly clear what's going on.